source/common/network/connection_impl.cc

```diff
@@ -106,10 +112,29 @@ void ConnectionImpl::close(ConnectionCloseType type) {
 
     closeSocket(ConnectionEvent::LocalClose);
   } else {
-    // TODO(mattklein123): We need a flush timer here. We might never get open socket window.
-    ASSERT(type == ConnectionCloseType::FlushWrite);
-    close_with_flush_ = true;
+    RELEASE_ASSERT(type == ConnectionCloseType::FlushWrite ||
+                       type == ConnectionCloseType::FlushWriteAndDelay,
+                   "");
```

Comment:
I'm using the RELEASE_ASSERT to provide runtime protection against bugs introduced in this logic, since they could lead to issues such as resource leaks and/or memory corruption.

I just re-read the style guide and it does mention that ASSERT should be the common case for this type of check, so I can certainly change it if that's the convention.

Reply:
Changed to ASSERT() per style guidelines.